modules/icgc-argo-workflows/prep_sample/resources/usr/bin/main.py

```diff
@@ -0,0 +1,355 @@
+      shutil.rmtree(args.outdir)
+    os.mkdir(args.outdir)
+
+    with open(args.metadata_json, 'r') as f:
+      song_analysis = json.load(f)
+
+    study_id = song_analysis['studyId']
+    donor_id = song_analysis['samples'][0]['donor']['donorId']
+    sample_id = song_analysis['samples'][0]['sampleId']
+    specimen_id = song_analysis['samples'][0]['specimen']['specimenId']
+    sex = 'XX' if song_analysis['samples'][0]['donor']['gender'] == 'Female' else 'XY' 
+    specimen_type = song_analysis['samples'][0]['specimen']['specimenType']
+    tumour_normal_designation = song_analysis['samples'][0]['specimen']['tumourNormalDesignation']
+    status = '0' if tumour_normal_designation == 'Normal' else '1'
+    analysis_type = song_analysis['analysisType']['name']
+    output_sample_sheet = f'{args.outdir}/{sample_id}_{analysis_type}_sample_sheet.csv'
+    
+
+    sample_sheet = dict()
+    if analysis_type == 'sequencing_experiment':
+      read_group_count = song_analysis['read_group_count']
+      filepair_map_to_readgroup = group_readgroup_by_filepair(song_analysis)
+
+      for fp in filepair_map_to_readgroup:
+        if filepair_map_to_readgroup[fp]['format'] == 'BAM':
+          # for bam just need fp[0] since fp[1] is either the same as fp[0] or None
+          sample_sheet = generate_fastqs_from_bam(filename_to_file(fp, args.input_files)[0],
+                                      filepair_map_to_readgroup[fp]['read_groups'],
+                                      args.cpus, sample_sheet, study_id, donor_id, sample_id, 
+                                      args.tempdir, args.outdir, specimen_id, specimen_type, 
+                                      tumour_normal_designation)
+        else: # FASTQ must be one read group
+          fq_pair = filename_to_file(fp, args.input_files)
+          fastq_pair = bunzip2(fq_pair)
+          rg = filepair_map_to_readgroup[fp]['read_groups'][0]
+          rg_id = rg['submitter_read_group_id']
+          rg_id_fn = readgroup_id_to_fname(rg_id, '', study_id, donor_id, sample_id)
+          file_r1_new = get_new_filename(fastq_pair[0], rg_id_fn, "R1", args.outdir)
+          os.symlink(os.path.abspath(fastq_pair[0]), file_r1_new)
+          if fastq_pair[1]:
+            file_r2_new = get_new_filename(fastq_pair[1], rg_id_fn, "R2", args.outdir)
+            os.symlink(os.path.abspath(fastq_pair[1]), file_r2_new)
+          else:
+            file_r2_new = 'No_File'
+
+          sample_sheet[rg_id] = {
+            'file_r1': file_r1_new,
+            'file_r2': file_r2_new
+          }
+        
+          # retrieve read_group_info from metadata
+          read_group_info = get_read_group_info(rg, study_id, donor_id, sample_id, specimen_id, specimen_type, tumour_normal_designation)
+
+          if read_group_info:
+            rg_kv = [ '@RG' ] + [ '%s:%s' % (k, v) for k, v in read_group_info.items() ]
+            rg_array = "\""+'\\t'.join(rg_kv)+"\""
```

Comment:
replace `"\""` with `"\'"`

Otherwise results in this output with `"""`:
```
analysis_type,study_id,patient,sex,status,sample,lane,fastq_1,fastq_2,read_group,single_end,read_group_count
sequencing_experiment,TEST-QA,DO263089,XX,1,SA624380,C0HVY.2,/Users/esu/Desktop/GitHub/icgc-argo-workflows/argo-modules/out/TEST-QA.DO263089.SA624380.C0HVY.2.8775eee1cacedc27428856591023d837_R1.fq.gz,/Users/esu/Desktop/GitHub/icgc-argo-workflows/argo-modules/out/TEST-QA.DO263089.SA624380.C0HVY.2.8775eee1cacedc27428856591023d837_R2.fq.gz,"""@RG\tID:C0HVY.2\tSM:SA624380\tLB:Pond-147580\tPU:74_8a\tPI:298\tCN:EXT\tPL:ILLUMINA\tPM:HiSeq 2000\tDT:2014-12-12\tDS:WGS|TEST-QA|SP224367|DO263089|Cell line - derived from tumour|Tumour""",False,3
```

Reply:
fix it.

Reply:
Same issue:
https://github.com/icgc-argo-workflows/argo-modules/blob/cb87db6e14bde3f8398cf93450f6b407f0237cc7/modules/icgc-argo-workflows/prep_sample/resources/usr/bin/main.py#L299